### PR TITLE
Add doc about MediaQuery to Chip

### LINKED ISSUE #21232: MediaQuery is required for Chips but there's no documentation about it?
See https://github.com/flutter/flutter/blob/4dbbf678874c473d8b25506019836a35fdc04a40/packages/flutter/lib/src/material/chip.dart#L1364 and https://docs.flutter.io/flutter/material/Chip-class.html

### PR DESCRIPTION
**Pull request by liyuqian**
Fixes https://github.com/flutter/flutter/issues/21232

### CODE REVIEW

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

**Review by goderbauer — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/material/chip.dart [hunk 1]
```diff
@@ -448,8 +448,9 @@ abstract class TappableChipAttributes {
 /// Supplying a non-null [onDeleted] callback will cause the chip to include a
 /// button for deleting the chip.
 ///
-/// Requires one of its ancestors to be a [Material] widget. The [label]
-/// and [clipBehavior] arguments must not be null.
+/// Its ancestors must include [Material], [MediaQuery], [Directionality], and
+/// [MaterialLocalizations]. The [label] and [clipBehavior] arguments must not
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/chip.dart]: You could add that typically all of these widgets are provided by [MaterialApp].

**Review by HansMuller — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/material/chip.dart (hunk 1 above)
Comment (liyuqian) [reply packages/flutter/lib/src/material/chip.dart]: Done. Thanks for the suggestion!

**Pull request merged**